src/components/SupplementalFiles/SupplementalFiles.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { useManifestState } from '../../context/manifest-context';
+import { getRenderingFiles, getSupplementingFiles } from '@Services/iiif-parser';
+import { fileDownload } from '@Services/utility-helpers';
+import './SupplementalFiles.scss';
+
+const SupplementalFiles = ({ itemHeading = "Item files", sectionHeading = "Section files", showHeading = true }) => {
+  const { manifest } = useManifestState();
+
+  const [manifestSupplementalFiles, setManifestSupplementalFiles] = React.useState();
+  const [canvasSupplementalFiles, setCanvasSupplementalFiles] = React.useState();
+  const [hasFiles, setHasFiles] = React.useState(false);
+
+  React.useEffect(() => {
+    if (manifest) {
+      let renderings = getRenderingFiles(manifest);
+
+      let manifestFiles = renderings.manifest;
+      setManifestSupplementalFiles(manifestFiles);
+
+      let annotations = getSupplementingFiles(manifest);
+      let canvasFiles = renderings.canvas;
+      canvasFiles.map((canvas, index) => canvas.files = canvas.files.concat(annotations[index].files));
+      canvasFiles = canvasFiles.filter(canvasFiles => canvasFiles.files.length > 0);
+      setCanvasSupplementalFiles(canvasFiles);
+
+      if (canvasFiles?.length > 0 && manifestFiles?.length > 0) {
+        setHasFiles(true);
+      }
```

Comment:
What do you think about converting this into a function instead of storing it in the state?  Would that even work?

Reply:
I guess it will work even without storing it in the state because the variables that determine its values are state variables which get updated when props change. Let me check.

Reply:
Yes, it works like that. I changed the code.